# **Introduction**

## **Chi-test (boroughs + svi)**
## **Bar-chart with svi as regression/scatterplot (boroughs first)**

source: https://www.atsdr.cdc.gov/place-health/media/pdfs/2024/10/SVI2022Documentation.pdf

source: https://www.atsdr.cdc.gov/place-health/php/svi/svi-interactive-map.html

Chi-test: with the whole dataset considered, the chi-test gives much more moderate and accurate result. There is no statistically strong association between high svi nta and high evi nta. (But if you move into a nta with higher-than-average svi, you are still 2.8 times more likely to get evicted compared to living in a lower than average svi nta). There is a statistically strong association between high black + hispanics nta with high evi nta. (and the odds are 3.81).

In [2]:
# !pip install geopandas folium matplotlib seaborn scipy
# !pip install esda
# !pip install splot
# !pip install geopandas contextily
# # for google colab, had to reinstall some pacakges.

In [ ]:
# !pip install geopandas folium matplotlib seaborn scipy esda splot

In [3]:
import pandas as pd
import geopandas as gpd
import numpy as np
import datetime as dt
import scipy

from sklearn.cluster import DBSCAN
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import silhouette_score
from shapely.geometry import Point
from sklearn.neighbors import NearestNeighbors

# visualization
import matplotlib.pyplot as plt
from matplotlib import colors as mcolors
import seaborn as sns
import folium
from folium.plugins import HeatMap
from folium import Marker
from folium.plugins import MarkerCluster
import plotly.express as px
import plotly.io as pio


import contextily as ctx
from scipy.stats import f_oneway
from sklearn.decomposition import PCA
from functools import reduce
from scipy.stats import chi2_contingency
import statsmodels.api as sm

# spatial statistics
from esda.moran import Moran
from esda.getisord import G_Local
from libpysal.weights import Queen, Rook

# system and utility
import warnings
import os
import io
from IPython.display import IFrame
from google.colab import files

from libpysal.weights import Queen, Rook
from esda.moran import Moran
import matplotlib.pyplot as plt
from splot.esda import moran_scatterplot

# suppress warnings
warnings.filterwarnings('ignore')

# inline
%matplotlib inline

In [4]:
pd.set_option('display.float_format', lambda x: '%.4f' % x)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

# **Step 1 Get the Eviction data**

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
# data source:
file_path1 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_normal_times.csv'
file_path2 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_covid.csv'

In [7]:
evictions_pre_post_raw = pd.read_csv(file_path1)
evictions_covid_raw = pd.read_csv(file_path2)
evictions_covid_raw.shape, evictions_pre_post_raw.shape
# 91 and 92 with normal time + one more analysis column regarding an svi item

((5386, 91), (66397, 92))

In [8]:
evictions_pre_post = evictions_pre_post_raw.copy()
evictions_covid = evictions_covid_raw.copy()

In [9]:
evictions_pre_post.head(2)

primary_key         bbl court_index_number  docket_number  \
0  *308072/22_5865  3037420029         *308072/22           5865   
1  *313639/23_5202  3057940012         *313639/23           5202   

  eviction_address eviction_apartment_number executed_date   borough  zipcode  \
0   356 MILLER AVE            1 AND BASEMENT    2024-12-04  BROOKLYN    11207   
1  710 61ST STREET                 2ND FLOOR    2024-03-04  BROOKLYN    11220   

          ejectment eviction/legal_possession  latitude  longitude  \
0  Not an Ejectment                Possession   40.6721   -73.8911   
1  Not an Ejectment                Possession   40.6359   -74.0119   

   community_board  council_district  census_tract      bin               nta  \
0           5.0000           37.0000     1152.0000  3083989     East New York   
1           7.0000           38.0000      118.0000  3143881  Sunset Park East   

   year month_year                      geometry  average_year_eviction_count  \
0  2024    2024-12  POINT (-73.891105 40.672121)                       0.8000   
1  2024    2024-03  POINT (-74.011883 40.635941)                       0.6000   

   yearbuilt bldgclass  numfloors  unitsres            ownername  bldgarea  \
0  1930.0000        C0     3.0000    3.0000       356 MILLER LLC 2700.0000   
1  1920.0000        B2     2.0000    2.0000  A.R.M. PARKING, LLC 1204.0000   

  building_type building_category  is_condo floor_category  \
0       pre-war           walk-up     False       low-rise   
1       pre-war        two-family     False       low-rise   

                     rent_era              architectural_style  \
0  Pre-1947, pre-rent-control  1921–1930, Art Deco Skyscrapers   
1  Pre-1947, pre-rent-control            1900–1920, Beaux-Arts   

                        economic_period residential_units_category  is_llc  \
0  1930-1945, great depression and WWII                  3-5 units    True   
1        Pre-1929, pre-great depression                     2-unit    True   

  building_size_category      size_quartile     decade   fips   e_totpop  \
0                  small        Q3 (50-75%)  1930-1939  11207 96801.0000   
1             very small  Q1 (smallest 25%)  1920-1929  11220 93008.0000   

   rpl_theme1  rpl_theme2  rpl_theme3  rpl_theme4  rpl_themes  ep_pov150  \
0      0.9788      0.9140      0.9808      0.9812      0.9839    33.9000   
1      0.9885      0.7635      0.9594      0.9179      0.9662    37.5000   

   ep_unemp  ep_nohsdp  ep_uninsur  ep_age65  ep_age17  ep_disabl  ep_limeng  \
0   11.1000    19.1000      6.0000   13.8000   22.5000    13.8000     5.3000   
1    7.5000    37.9000     11.6000   13.1000   25.4000     8.4000    40.2000   

   ep_noveh  ep_crowd  ep_hburd  ep_afam  ep_hisp  ep_asian  ep_aian  ep_nhpi  \
0   57.8000    9.1000   44.7000  55.9000  32.8000    1.5000   0.0000   0.0000   
1   61.7000   23.7000   43.6000   1.7000  40.9000   40.7000   0.4000   0.0000   

   ep_twomore  ep_otherrace  ep_minrty  ep_white  invalid_zip svi_quartile  \
0      2.9000        1.6000    94.7000    5.3000        False           Q3   
1      1.2000        0.2000    85.0000   15.0000        False           Q3   

     svi_group  air_quality  animal_issues  appliances  building_exterior  \
0  medium-high       0.0000         0.0000      1.0000             0.0000   
1  medium-high       0.0000         0.0000      0.0000             0.0000   

   doors_windows  electrical_issues  elevator_issues  floors_stairs  \
0         1.0000             2.0000           0.0000         0.0000   
1         0.0000             0.0000           0.0000         0.0000   

   general_complaints  graffiti_posting  heat_hot_water  homeless_issues  \
0              1.0000            0.0000          3.0000           0.0000   
1              0.0000            0.0000          1.0000           0.0000   

   noise_complaints  other_issues  pest_issues  plumbing_issues  \
0            0.0000        0.0000       0.0000           3.0000   
1            0.0

In [10]:
evictions_covid.head(2)

primary_key         bbl court_index_number  docket_number  \
0   004123/20_209969  2032140141          004123/20         209969   
1  0050153/20_106030  4031560133         0050153/20         106030   

    eviction_address eviction_apartment_number executed_date borough  zipcode  \
0   2541 A GRAND AVE                   ROOM 3B    2022-08-22   BRONX    10468   
1  98-05 67TH AVENUE                       12F    2022-04-14  QUEENS    11375   

          ejectment eviction/legal_possession  latitude  longitude  \
0  Not an Ejectment                Possession   40.8654   -73.9013   
1  Not an Ejectment                Possession   40.7242   -73.8556   

   community_board  council_district  census_tract      bin  \
0           7.0000           14.0000      265.0000  2113173   
1           6.0000           29.0000    71306.0000  4074666   

                   nta  year month_year                      geometry  \
0  Kingsbridge Heights  2022    2022-08  POINT (-73.901317 40.865396)   
1         Forest Hills  2022    2022-04  POINT (-73.855552 40.724241)   

   average_year_eviction_count  yearbuilt bldgclass  numfloors  unitsres  \
0                       0.2000  2004.0000        C0     3.0000    3.0000   
1                       0.2000  1960.0000        D3    13.0000  181.0000   

                                ownername    bldgarea building_type  \
0                            MONJU SARKER   3420.0000      post-war   
1  MARSEILLES LEASING LIMITED PARTNERSHIP 177710.0000      post-war   

  building_category  is_condo floor_category                         rent_era  \
0           walk-up     False       low-rise  1994–Present, vacancy decontrol   
1          elevator     False      high-rise          1947–1969, rent-control   

                                 architectural_style  \
0                     2001-present, New Architecture   
1  1951–1980, the International Style, Alternativ...   

                     economic_period residential_units_category  is_llc  \
0  1991–2008, modern economic growth                  3-5 units   False   
1   1946–1975, pst war economic boom                 100+ units   False   

  building_size_category     size_quartile     decade   fips   e_totpop  \
0           medium-small  Q4 (largest 25%)  2000-2009  10468 81397.0000   
1                   mega  Q4 (largest 25%)  1960-1969  11375 75212.0000   

   rpl_theme1  rpl_theme2  rpl_theme3  rpl_theme4  rpl_themes  ep_pov150  \
0      0.9954      0.9407      0.9870      0.9470      0.9874    39.5000   
1      0.4759      0.5698      0.8789      0.8057      0.7322    12.0000   

   ep_unemp  ep_nohsdp  ep_uninsur  ep_age65  ep_age17  ep_disabl  ep_limeng  \
0   11.6000    28.3000      9.2000   11.2000   26.4000    12.2000    26.9000   
1    4.8000     6.1000      3.7000   20.4000   18.0000    10.5000     7.9000   

   ep_noveh  ep_crowd  ep_hburd  ep_afam  ep_hisp  ep_asian  ep_aian  ep_nhpi  \
0   71.8000   19.2000   56.7000  15.6000  78.0000    2.3000   0.0000   0.0000   
1   41.9000    5.8000   25.4000   2.7000  16.4000   28.5000   0.1000   0.0000   

   ep_twomore  ep_otherrace  ep_minrty  ep_white  invalid_zip svi_quartile  \
0      0.5000        0.5000    96.9000    3.1000        False           Q3   
1      4.6000        0.7000    53.0000   47.0000        False     Q1 (Low)   

   air_quality  animal_issues  appliances  building_exterior  doors_windows  \
0       0.0000         0.0000      0.0000             0.0000         3.0000   
1       0.0000         2.0000      0.0000             0.0000         0.0000   

   electrical_issues  elevator_issues  floors_stairs  general_complaints  \
0             0.0000           0.0000         2.0000              0.0000   
1             0.0000           0.0000         0.0000              2.0000   

   graffiti_posting  heat_hot_water  homeless_issues  noise_complaints  \
0            0.0000          1.0000           0.0000            2.0000   
1            0.0000         62.0000           0.0000         

In [11]:
evictions_pre_post.columns, \
evictions_covid.columns, \
evictions_pre_post.shape, \
evictions_covid.shape

(Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
        'latitude', 'longitude', 'community_board', 'council_district',
        'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
        'average_year_eviction_count', 'yearbuilt', 'bldgclass', 'numfloors',
        'unitsres', 'ownername', 'bldgarea', 'building_type',
        'building_category', 'is_condo', 'floor_category', 'rent_era',
        'architectural_style', 'economic_period', 'residential_units_category',
        'is_llc', 'building_size_category', 'size_quartile', 'decade', 'fips',
        'e_totpop', 'rpl_theme1', 'rpl_theme2', 'rpl_theme3', 'rpl_theme4',
        'rpl_themes', 'ep_pov150', 'ep_unemp', 'ep_nohsdp', 'ep_uninsur',
        'ep_age65', 'ep_age17', 'ep_disabl', 'ep_limeng', 'ep_noveh',
        'ep_crowd', 'ep_hburd', 'ep_afam', 'ep_hisp

In [12]:
link = '/content/drive/My Drive/X999/svi_cleaned.csv'

In [13]:
svi_df = pd.read_csv(link)
svi_df.head(2)

fips     location  area_sqmi  e_totpop  m_totpop   e_hu  m_hu   e_hh  \
0  10001  ZCTA5 10001     0.6238     27004      1827  16975   831  14375   
1  10002  ZCTA5 10002     0.8223     76518      2894  39094  1241  36028   

   m_hh  e_pov150  m_pov150  e_unemp  m_unemp  e_hburd  m_hburd  e_nohsdp  \
0   782      5248       797      761      266     3314      531      1930   
1  1326     27908      2853     2833      574    14688     1367     18301   

   m_nohsdp  e_uninsur  m_uninsur  e_age65  m_age65  e_age17  m_age17  \
0       534        831        289     3428      432     2694      643   
1      1376       4074        766    17681     1287    10028     1549   

   e_disabl  m_disabl  e_sngpnt  m_sngpnt  e_limeng  m_limeng  e_minrty  \
0      2310       499       501       215      1381       405     13460   
1      9896      1062      2211       499     18393      1640     56964   

   m_minrty  e_munit  m_munit  e_mobile  m_mobile  e_crowd  m_crowd  e_noveh  \
0      2305    15840      898        15        23      389      135    12285   
1      3226    35725     1677        16        28     2461      449    29828   

   m_noveh  e_groupq  m_groupq  ep_pov150  mp_pov150  ep_unemp  mp_unemp  \
0      840      2213       218    20.3000     2.7000    4.3000    1.5000   
1     1403      2090        39    36.8000     3.5000    7.6000    1.4000   

   ep_hburd  mp_hburd  ep_nohsdp  mp_nohsdp  ep_uninsur  mp_uninsur  ep_age65  \
0   23.1000    3.5000     9.1000     2.4000      3.1000      1.0000   12.7000   
1   40.8000    3.5000    30.0000     2.0000      5.4000      1.0000   23.1000   

   mp_age65  ep_age17  mp_age17  ep_disabl  mp_disabl  ep_sngpnt  mp_sngpnt  \
0    1.6000   10.0000    2.1000     8.6000     1.9000     3.5000     1.5000   
1    1.7000   13.1000    1.8000    13.0000     1.4000     6.1000     1.4000   

   ep_limeng  mp_limeng  ep_minrty  mp_minrty  ep_munit  mp_munit  ep_mobile  \
0     5.3000     1.5000    49.8000     7.8000   93.3000    2.7000     0.1000   
1    24.7000     2.0000    74.4000     3.1000   91.4000    3.2000     0.0000   

   mp_mobile  ep_crowd  mp_crowd  ep_noveh  mp_noveh  ep_groupq  mp_groupq  \
0     0.1000    2.7000    0.9000   85.5000    2.8000     8.2000     0.6000   
1     0.1000    6.8000    1.2000   82.8000    1.8000     2.7000     0.1000   

   epl_pov150  epl_unemp  epl_hburd  epl_nohsdp  epl_uninsur  spl_theme1  \
0      0.6108     0.4574     0.5573      0.5902       0.4436      2.6593   
1      0.9148     0.7946     0.9219      0.9741       0.7207      4.3261   

   rpl_theme1  epl_age65  epl_age17  epl_disabl  epl_sngpnt  epl_limeng  \
0      0.5688     0.1420     0.1161      0.1891      0.4707      0.8777   
1      0.9639     0.7296     0.1831      0.5186      0.7390      0.9944   

   spl_theme2  rpl_theme2  epl_minrty  spl_theme3  rpl_theme3  epl_munit  \
0      1.7956      0.1692      0.8670      0.8670      0.8670     0.9853   
1      3.1647      0.8781      0.9369      0.9369      0.9369     0.9790   

   epl_mobile  epl_crowd  epl_noveh  epl_groupq  spl_theme4  rpl_theme4  \
0      0.2710     0.7402     0.9949      0.9104      3.9018      0.9806   
1      0.0000     0.9105     0.9915      0.7730      3.6540      0.9254   

   spl_themes  rpl_themes  f_pov150  f_unemp  f_hburd  f_nohsdp  f_uninsur  \
0      9.2237      0.7414         0        0        0         0          0   
1     12.0817      0.9656         1        0        1         1          0   

   f_theme1  f_age65  f_age17  f_disabl  f_sngpnt  f_limeng  f_theme2  \
0         0        0        0         0         0         0         0   
1         3        0        0         0         0         1         1   

   f_minrty  f_theme3  f_munit  f_mobile  f_crowd  f_noveh  f_groupq  \
0         0         0        1         0        0        1         1   
1         1         1        1         0        1        1         0   

   f_theme4  f_total  e_daypop  e_noint  m_noint  e_afam  m_afam  e_hisp  \
0         3

In [14]:
svi_df.shape

(204, 153)

In [15]:
# list(svi_df.columns)

In [16]:
svi_df.ep_nhpi.unique()

array([ 0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,  8.00e-01,
        1.20e+00,  5.00e-01,  4.00e-01])

# **Step 2: SVI items**

q quick double check

In [17]:
link = "/content/drive/My Drive/X999/NewYork_ZCTA.csv"

In [18]:
svi_raw = pd.read_csv(link)
svi_raw.head(2)

ST     STATE ST_ABBR   FIPS     LOCATION  AREA_SQMI  E_TOTPOP  M_TOTPOP  \
0  36  New York      NY   6390  ZCTA5 06390     4.0467        53        39   
1  36  New York      NY  10001  ZCTA5 10001     0.6238     27004      1827   

    E_HU  M_HU   E_HH  M_HH  E_POV150  M_POV150  E_UNEMP  M_UNEMP  E_HBURD  \
0    253    49     19    19        17        16        0       13        9   
1  16975   831  14375   782      5248       797      761      266     3314   

   M_HBURD  E_NOHSDP  M_NOHSDP  E_UNINSUR  M_UNINSUR  E_AGE65  M_AGE65  \
0       26         0        13         27         34        0       13   
1      531      1930       534        831        289     3428      432   

   E_AGE17  M_AGE17  E_DISABL  M_DISABL  E_SNGPNT  M_SNGPNT  E_LIMENG  \
0        6       11        31        33         0        18         9   
1     2694      643      2310       499       501       215      1381   

   M_LIMENG  E_MINRTY  M_MINRTY  E_MUNIT  M_MUNIT  E_MOBILE  M_MOBILE  \
0        53        20        51        0       18         4         5   
1       405     13460      2305    15840      898        15        23   

   E_CROWD  M_CROWD  E_NOVEH  M_NOVEH  E_GROUPQ  M_GROUPQ  EP_POV150  \
0        0       18        0       13        17        16    32.1000   
1      389      135    12285      840      2213       218    20.3000   

   MP_POV150  EP_UNEMP  MP_UNEMP  EP_HBURD  MP_HBURD  EP_NOHSDP  MP_NOHSDP  \
0    18.8000    0.0000   52.7000   47.4000  100.0000     0.0000    51.4000   
1     2.7000    4.3000    1.5000   23.1000    3.5000     9.1000     2.4000   

   EP_UNINSUR  MP_UNINSUR  EP_AGE65  MP_AGE65  EP_AGE17  MP_AGE17  EP_DISABL  \
0     50.9000     45.7000    0.0000   45.2000   11.3000   19.0000    58.5000   
1      3.1000      1.0000   12.7000    1.6000   10.0000    2.1000     8.6000   

   MP_DISABL  EP_SNGPNT  MP_SNGPNT  EP_LIMENG  MP_LIMENG  EP_MINRTY  \
0    41.4000     0.0000    94.7000    17.0000    99.2000    37.7000   
1     1.9000     3.5000     1.5000     5.3000     1.5000    49.8000   

   MP_MINRTY  EP_MUNIT  MP_MUNIT  EP_MOBILE  MP_MOBILE  EP_CROWD  MP_CROWD  \
0    92.1000    0.0000    7.1000     1.6000     2.0000    0.0000   94.7000   
1     7.8000   93.3000    2.7000     0.1000     0.1000    2.7000    0.9000   

   EP_NOVEH  MP_NOVEH  EP_GROUPQ  MP_GROUPQ  EPL_POV150  EPL_UNEMP  EPL_HBURD  \
0    0.0000   75.5000    32.1000    18.8000      0.8790     0.0000     0.9635   
1   85.5000    2.8000     8.2000     0.6000      0.6108     0.4574     0.5573   

   EPL_NOHSDP  EPL_UNINSUR  SPL_THEME1  RPL_THEME1  EPL_AGE65  EPL_AGE17  \
0      0.0000       0.9960      2.8385      0.6342     0.0000     0.1408   
1      0.5902       0.4436      2.6593      0.5688     0.1420     0.1161   

   EPL_DISABL  EPL_SNGPNT  EPL_LIMENG  SPL_THEME2  RPL_THEME2  EPL_MINRTY  \
0      0.9944      0.0000      0.9775      2.1127      0.3009      0.8062   
1      0.1891      0.4707      0.8777      1.7956      0.1692      0.8670   

   SPL_THEME3  RPL_THEME3  EPL_MUNIT  EPL_MOBILE  EPL_CROWD  EPL_NOVEH  \
0      0.8062      0.8062     0.0000      0.4654     0.0000     0.0000   
1      0.8670      0.8670     0.9853      0.2710     0.7402     0.9949   

   EPL_GROUPQ  SPL_THEME4  RPL_THEME4  SPL_THEMES  RPL_THEMES  F_POV150  \
0      0.9735      1.4389      0.2205      7.1963      0.4192         0   
1      0.9104      3.9018      0.9806      9.2237      0.7414         0   

   F_UNEMP  F_HBURD  F_NOHSDP  F_UNINSUR  F_THEME1  F_AGE65  F_AGE17  \
0        0        1         0          1         2        0        0   
1        0        0         0          0         0        0        0   

   F_DISABL  F_SNGPNT  F_LIMENG  F_THEME2  F_MINRTY  F_THEME3  F_MUNIT  \
0         1         0         1         2         0         0        0   
1         0         0         0         0         0         0        1   

   F_MOBILE  F_CROWD  F_NOVEH  F_GROUPQ  F_THEME4  F_TOTAL  E_DAYPOP  E_NOINT  \
0         0        0        0         1         1    

In [19]:
def is_nyc_zipcode(zipcode):
    zip_int = int(zipcode) if isinstance(zipcode, str) else zipcode

    # Manhattan: 10001-10282
    if 10001 <= zip_int <= 10282:
        return True
    # addition Manhattan: 10300-10499
    if 10300 <= zip_int <= 10499:
        return True
    # Bronx: 10451-10475
    if 10451 <= zip_int <= 10475:
        return True
    # Brooklyn: 11201-11256
    if 11201 <= zip_int <= 11256:
        return True
    # Queens: 11351-11436, 11101-11109
    if (11351 <= zip_int <= 11436) or (11101 <= zip_int <= 11109):
        return True
    # Staten Island: 10301-10314
    if 10301 <= zip_int <= 10314:
        return True
    # additional Queens ZIPs
    if zip_int in [11004, 11005, 11411, 11412, 11413, 11418, 11419, 11420, 11421, 11422, 11423, 11426, 11427, 11428, 11429]:
        return True
    return False

In [20]:
nyc_df = svi_raw[svi_raw['FIPS'].apply(is_nyc_zipcode)]

In [21]:
nyc_df.shape

(204, 156)

In [22]:
nyc_df.EP_NHPI.unique()
# interesting, so there was an error in the source data

array([ 0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,  8.00e-01,
        1.20e+00,  5.00e-01,  4.00e-01])

In [23]:
svi_raw.EP_NHPI.unique()

array([ 1.51e+01,  0.00e+00,  1.00e-01,  3.00e-01,  2.00e-01, -9.99e+02,
        8.00e-01,  1.20e+00,  5.00e-01,  4.00e-01,  1.10e+00,  7.00e-01,
        1.50e+00,  1.80e+00,  9.00e-01,  6.00e-01,  1.40e+00,  2.10e+00,
        2.30e+00,  1.00e+00])

In [24]:
-9.99e+02, 0.00e+00, 9.00e-01, 2.00e-01

(-999.0, 0.0, 0.9, 0.2)

# **Step 3: All boroughs and their eviction rates**

In [25]:
evictions_pre_post_mean = evictions_pre_post[['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']].mean()
evictions_pre_post_mean
# may need to merge ep_twomore and ep_otherrance toegther

ep_afam        29.2346
ep_asian        8.9040
ep_hisp        38.7307
ep_nhpi         0.0115
ep_white       19.0236
ep_twomore      2.8025
ep_otherrace    1.0239
dtype: float64

In [26]:
evictions_pre_post_mean = evictions_pre_post_mean.reset_index()

In [27]:
evictions_pre_post_mean.rename(columns = {'index':'race_svi', 0: "racial percentage"}, inplace=True)

In [28]:
evictions_pre_post_mean

race_svi  racial percentage
0       ep_afam            29.2346
1      ep_asian             8.9040
2       ep_hisp            38.7307
3       ep_nhpi             0.0115
4      ep_white            19.0236
5    ep_twomore             2.8025
6  ep_otherrace             1.0239

In [29]:
# type(evictions_pre_post_mean)
# so this is correct

In [30]:
# evictions_pre_post.columns

In [160]:
neighbor_evictions = evictions_pre_post.groupby('nta').agg({'average_year_eviction_count': 'mean', 'borough': 'first'}).reset_index()
neighbor_evictions.sort_values('average_year_eviction_count', ascending=False, inplace=True)
neighbor_evictions

nta  \
185                            park-cemetery-etc-Bronx   
143                               Seagate-Coney Island   
36                                              Corona   
163                  University Heights-Morris Heights   
125                                 Park Slope-Gowanus   
75                       Grymes Hill-Clifton-Fox Hills   
106                                 Morrisania-Melrose   
19                                           Bronxdale   
21                                         Brownsville   
31                                  Claremont-Bathgate   
175                              Westchester-Unionport   
7                    Battery Park City-Lower Manhattan   
62                                       Fordham South   
162                            Turtle Bay-East Midtown   
119                              Oakwood-Oakwood Beach   
128                                     Pelham Parkway   
44                    East Concourse-Concourse Village   
108                                         Mount Hope   
66                    Ft. Totten-Bay Terrace-Clearview   
101                     Melrose South-Mott Haven North   
52                                        East Tremont   
61                                            Flushing   
182                                 Woodlawn-Wakefield   
77                                          Highbridge   
87                                         Kew Gardens   
39                                 Crown Heights South   
171                                     West Concourse   
27                   Central Harlem North-Polo Grounds   
150                          Springfield Gardens North   
34                                          Co-op City   
13                                             Belmont   
88                                   Kew Gardens Hills   
103                              Midtown-Midtown South   
37                                   Crotona Park East   
33                                        Clinton Hill   
11                          Bedford Park-Fordham North   
132                  Prospect Lefferts Gardens-Wingate   
82                                         Hunts Point   
115                North Riverdale-Fieldston-Riverdale   
166                              Van Cortlandt Village   
172                             West Farms-Bronx River   
146                                 Soundview-Bruckner   
178                           Williamsbridge-Olinville   
147    Soundview-Castle Hill-Clason Point-Harding Park   
117                                            Norwood   
170                                      West Brighton   
94                                            Longwood   
43                                       Dyker Heights   
17                             Briarwood-Jamaica Hills   
155                                      Starrett City   
120                                         Ocean Hill   
152                         Spuyten Duyvil-Kingsbridge   
42                Douglas Manor-Douglaston-Little Neck   
109                                        Murray Hill   
89                                 Kingsbridge Heights   
64                                         Fort Greene   
126                                        Parkchester   
99   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
48                                   East Harlem North   
54                                   East Williamsburg   
46                              East Flatbush-Farragut   
84                                             Jamaica   
80          Hudson Yards-Chelsea-Flatiron-Union Square   
23                                      Bushwick South   
41    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
136                                          Rego Park   
50                                       East New York   
49                                   East Harlem South   
98                                  Marble Hill-I

In [163]:
man_nta = {
    'nta': neighbor_evictions['nta'].unique(),
    'eviction_rates': neighbor_evictions['average_year_eviction_count']
}

man_nta_df = pd.DataFrame(man_nta)
man_nta_df

nta  eviction_rates
185                            park-cemetery-etc-Bronx          6.4667
143                               Seagate-Coney Island          4.3904
36                                              Corona          4.2597
163                  University Heights-Morris Heights          3.5051
125                                 Park Slope-Gowanus          2.3725
75                       Grymes Hill-Clifton-Fox Hills          2.3360
106                                 Morrisania-Melrose          2.2550
19                                           Bronxdale          2.2396
21                                         Brownsville          2.0909
31                                  Claremont-Bathgate          1.9783
175                              Westchester-Unionport          1.9408
7                    Battery Park City-Lower Manhattan          1.7746
62                                       Fordham South          1.6845
162                            Turtle Bay-East Midtown          1.6833
119                              Oakwood-Oakwood Beach          1.6722
128                                     Pelham Parkway          1.6411
44                    East Concourse-Concourse Village          1.6058
108                                         Mount Hope          1.5958
66                    Ft. Totten-Bay Terrace-Clearview          1.5800
101                     Melrose South-Mott Haven North          1.5741
52                                        East Tremont          1.5485
61                                            Flushing          1.5223
182                                 Woodlawn-Wakefield          1.5069
77                                          Highbridge          1.4761
87                                         Kew Gardens          1.4738
39                                 Crown Heights South          1.4379
171                                     West Concourse          1.4167
27                   Central Harlem North-Polo Grounds          1.4098
150                          Springfield Gardens North          1.4075
34                                          Co-op City          1.4012
13                                             Belmont          1.3974
88                                   Kew Gardens Hills          1.3964
103                              Midtown-Midtown South          1.3892
37                                   Crotona Park East          1.3511
33                                        Clinton Hill          1.3447
11                          Bedford Park-Fordham North          1.3362
132                  Prospect Lefferts Gardens-Wingate          1.3238
82                                         Hunts Point          1.3140
115                North Riverdale-Fieldston-Riverdale          1.3126
166                              Van Cortlandt Village          1.3000
172                             West Farms-Bronx River          1.2898
146                                 Soundview-Bruckner          1.2836
178                           Williamsbridge-Olinville          1.2645
147    Soundview-Castle Hill-Clason Point-Harding Park          1.2506
117                                            Norwood          1.2479
170                                      West Brighton          1.2430
94                                            Longwood          1.1946
43                                       Dyker Heights          1.1926
17                             Briarwood-Jamaica Hills          1.1921
155                                      Starrett City          1.1795
120                                         Ocean Hill          1.1774
152                         Spuyten Duyvil-Kingsbridge          1.1730
42                Douglas Manor-Douglaston-Little Neck          1.1280
109                                        Murray Hill          1.0903
89                                 Kingsbridge Heights          1.0777
64                                         Fort Greene          1.0685
126 

In [164]:
avg_per_nta = neighbor_evictions.average_year_eviction_count.mean()
avg_per_nta
# per building, per year, that's why it is similar to the borough one too

np.float64(0.9049163589883761)

### **Step 3.2. All neighborhoods racial composite**

In [170]:
race_columns = ['ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace']
racial_avg_all = evictions_pre_post.groupby('nta')[['average_year_eviction_count', 'ep_afam', 'ep_asian', 'ep_hisp', 'ep_nhpi', 'ep_white', 'ep_twomore', 'ep_otherrace', 'rpl_themes']].mean()
racial_avg_all.sort_values('average_year_eviction_count', ascending=False, inplace=True)
racial_avg_all.reset_index(inplace=True)
racial_avg_all

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

In [166]:
racial_avg_all.T

nta                          park-cemetery-etc-Bronx  Seagate-Coney Island  \
average_year_eviction_count                   6.4667                4.3904   
ep_afam                                      28.6000               20.1385   
ep_asian                                      6.2000                9.8270   
ep_hisp                                      52.9000               19.4159   
ep_nhpi                                       0.0000                0.0000   
ep_white                                      8.7000               47.3927   
ep_twomore                                    2.3000                2.9144   
ep_otherrace                                  0.7000                0.0159   
rpl_themes                                    0.9925                0.9911   

nta                          Corona  University Heights-Morris Heights  \
average_year_eviction_count  4.2597                             3.5051   
ep_afam                      7.3968                            24.4407   
ep_asian                    11.3050                             1.6079   
ep_hisp                     75.7668                            69.3353   
ep_nhpi                      0.0000                             0.0000   
ep_white                     4.4275                             2.0544   
ep_twomore                   0.5039                             1.8489   
ep_otherrace                 0.4997                             0.7046   
rpl_themes                   0.9632                             0.9979   

nta                          Park Slope-Gowanus  \
average_year_eviction_count              2.3725   
ep_afam                                  7.4853   
ep_asian                                 9.5010   
ep_hisp                                 16.3461   
ep_nhpi                                  0.0000   
ep_white                                60.6373   
ep_twomore                               5.4206   
ep_otherrace                             0.5765   
rpl_themes                               0.6327   

nta                          Grymes Hill-Clifton-Fox Hills  \
average_year_eviction_count                         2.3360   
ep_afam                                            25.1149   
ep_asian                                           11.6023   
ep_hisp                                            22.3858   
ep_nhpi                                             0.0000   
ep_white                                           38.7568   
ep_twomore                                          1.5584   
ep_otherrace                                        0.4703   
rpl_themes                                          0.9444   

nta                          Morrisania-Melrose  Bronxdale  Brownsville  \
average_year_eviction_count              2.2550     2.2396       2.0909   
ep_afam                                 35.4504    29.4301      70.0757   
ep_asian                                 0.6731     6.1401       0.8424   
ep_hisp                                 59.1572    52.1306      19.3509   
ep_nhpi                                  0.0000     0.0000       0.0002   
ep_white                                 2.5927     8.6984       3.6573   
ep_twomore                               1.5814     2.2915       5.3733   
ep_otherrace                             0.3022     0.7095       0.6034   
rpl_themes                               0.9965     0.9909       0.9933   

nta                          Claremont-Bathgate  Westchester-Unionport  \
average_year_eviction_count              1.9783                 1.9408   
ep_afam                                 30.5182                22.5381   
ep_asian                                 1.0987                10.2746   
ep_hisp                                 63.2627                55.1108   
ep_nhpi                                  0.0000                 0.0257   
ep_white                                 3.0336                 8.5335   
ep_twomore                               1.4791                 1.8548   
ep_

## **This is for the race composites and neighborhoods bar char use**

# **Step 4 We want a dataframe that has neighborhoods as columns and average_year_eviction_count as rows and contents**

In [171]:
evi_svi_df = evictions_pre_post.groupby('nta')[['average_year_eviction_count','rpl_themes']].mean()
evi_svi_df.sort_values('average_year_eviction_count', ascending=False, inplace=True)
evi_svi_df

average_year_eviction_count  \
nta                                                                               
park-cemetery-etc-Bronx                                                  6.4667   
Seagate-Coney Island                                                     4.3904   
Corona                                                                   4.2597   
University Heights-Morris Heights                                        3.5051   
Park Slope-Gowanus                                                       2.3725   
Grymes Hill-Clifton-Fox Hills                                            2.3360   
Morrisania-Melrose                                                       2.2550   
Bronxdale                                                                2.2396   
Brownsville                                                              2.0909   
Claremont-Bathgate                                                       1.9783   
Westchester-Unionport                                                    1.9408   
Battery Park City-Lower Manhattan                                        1.7746   
Fordham South                                                            1.6845   
Turtle Bay-East Midtown                                                  1.6833   
Oakwood-Oakwood Beach                                                    1.6722   
Pelham Parkway                                                           1.6411   
East Concourse-Concourse Village                                         1.6058   
Mount Hope                                                               1.5958   
Ft. Totten-Bay Terrace-Clearview                                         1.5800   
Melrose South-Mott Haven North                                           1.5741   
East Tremont                                                             1.5485   
Flushing                                                                 1.5223   
Woodlawn-Wakefield                                                       1.5069   
Highbridge                                                               1.4761   
Kew Gardens                                                              1.4738   
Crown Heights South                                                      1.4379   
West Concourse                                                           1.4167   
Central Harlem North-Polo Grounds                                        1.4098   
Springfield Gardens North                                                1.4075   
Co-op City                                                               1.4012   
Belmont                                                                  1.3974   
Kew Gardens Hills                                                        1.3964   
Midtown-Midtown South                                                    1.3892   
Crotona Park East                                                        1.3511   
Clinton Hill                                                             1.3447   
Bedford Park-Fordham North                                               1.3362   
Prospect Lefferts Gardens-Wingate                                        1.3238   
Hunts Point                                                              1.3140   
North Riverdale-Fieldston-Riverdale                                      1.3126   
Van Cortlandt Village                                                    1.3000   
West Farms-Bronx River                                                   1.2898   
Soundview-Bruckner                                                       1.2836   
Williamsbridge-Olinville                                                 1.2645   
Soundview-Castle Hill-Clason Point-Harding Park                          1.2506   
Norwood                                                                  1.2479   
West Brighton                                                            1.2430   
Longwood                                                            

In [145]:
evi_svi_df.reset_index(inplace=True)
evi_svi_df

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

In [146]:
avg_per_nta

np.float64(0.9049163589883761)

In [147]:
average_svi = 0.80198
# https://www.atsdr.cdc.gov/place-health/php/svi/svi-interactive-map.html

In [202]:
evi_svi_df['above_eviction_average'] = evi_svi_df['average_year_eviction_count'] > avg_per_nta
evi_svi_df

average_year_eviction_count  \
nta                                                                               
park-cemetery-etc-Bronx                                                  6.4667   
Seagate-Coney Island                                                     4.3904   
Corona                                                                   4.2597   
University Heights-Morris Heights                                        3.5051   
Park Slope-Gowanus                                                       2.3725   
Grymes Hill-Clifton-Fox Hills                                            2.3360   
Morrisania-Melrose                                                       2.2550   
Bronxdale                                                                2.2396   
Brownsville                                                              2.0909   
Claremont-Bathgate                                                       1.9783   
Westchester-Unionport                                                    1.9408   
Battery Park City-Lower Manhattan                                        1.7746   
Fordham South                                                            1.6845   
Turtle Bay-East Midtown                                                  1.6833   
Oakwood-Oakwood Beach                                                    1.6722   
Pelham Parkway                                                           1.6411   
East Concourse-Concourse Village                                         1.6058   
Mount Hope                                                               1.5958   
Ft. Totten-Bay Terrace-Clearview                                         1.5800   
Melrose South-Mott Haven North                                           1.5741   
East Tremont                                                             1.5485   
Flushing                                                                 1.5223   
Woodlawn-Wakefield                                                       1.5069   
Highbridge                                                               1.4761   
Kew Gardens                                                              1.4738   
Crown Heights South                                                      1.4379   
West Concourse                                                           1.4167   
Central Harlem North-Polo Grounds                                        1.4098   
Springfield Gardens North                                                1.4075   
Co-op City                                                               1.4012   
Belmont                                                                  1.3974   
Kew Gardens Hills                                                        1.3964   
Midtown-Midtown South                                                    1.3892   
Crotona Park East                                                        1.3511   
Clinton Hill                                                             1.3447   
Bedford Park-Fordham North                                               1.3362   
Prospect Lefferts Gardens-Wingate                                        1.3238   
Hunts Point                                                              1.3140   
North Riverdale-Fieldston-Riverdale                                      1.3126   
Van Cortlandt Village                                                    1.3000   
West Farms-Bronx River                                                   1.2898   
Soundview-Bruckner                                                       1.2836   
Williamsbridge-Olinville                                                 1.2645   
Soundview-Castle Hill-Clason Point-Harding Park                          1.2506   
Norwood                                                                  1.2479   
West Brighton                                                            1.2430   
Longwood                                                            

In [203]:
evi_svi_df['above_svi_average'] = evi_svi_df['rpl_themes'] > average_svi
evi_svi_df

average_year_eviction_count  \
nta                                                                               
park-cemetery-etc-Bronx                                                  6.4667   
Seagate-Coney Island                                                     4.3904   
Corona                                                                   4.2597   
University Heights-Morris Heights                                        3.5051   
Park Slope-Gowanus                                                       2.3725   
Grymes Hill-Clifton-Fox Hills                                            2.3360   
Morrisania-Melrose                                                       2.2550   
Bronxdale                                                                2.2396   
Brownsville                                                              2.0909   
Claremont-Bathgate                                                       1.9783   
Westchester-Unionport                                                    1.9408   
Battery Park City-Lower Manhattan                                        1.7746   
Fordham South                                                            1.6845   
Turtle Bay-East Midtown                                                  1.6833   
Oakwood-Oakwood Beach                                                    1.6722   
Pelham Parkway                                                           1.6411   
East Concourse-Concourse Village                                         1.6058   
Mount Hope                                                               1.5958   
Ft. Totten-Bay Terrace-Clearview                                         1.5800   
Melrose South-Mott Haven North                                           1.5741   
East Tremont                                                             1.5485   
Flushing                                                                 1.5223   
Woodlawn-Wakefield                                                       1.5069   
Highbridge                                                               1.4761   
Kew Gardens                                                              1.4738   
Crown Heights South                                                      1.4379   
West Concourse                                                           1.4167   
Central Harlem North-Polo Grounds                                        1.4098   
Springfield Gardens North                                                1.4075   
Co-op City                                                               1.4012   
Belmont                                                                  1.3974   
Kew Gardens Hills                                                        1.3964   
Midtown-Midtown South                                                    1.3892   
Crotona Park East                                                        1.3511   
Clinton Hill                                                             1.3447   
Bedford Park-Fordham North                                               1.3362   
Prospect Lefferts Gardens-Wingate                                        1.3238   
Hunts Point                                                              1.3140   
North Riverdale-Fieldston-Riverdale                                      1.3126   
Van Cortlandt Village                                                    1.3000   
West Farms-Bronx River                                                   1.2898   
Soundview-Bruckner                                                       1.2836   
Williamsbridge-Olinville                                                 1.2645   
Soundview-Castle Hill-Clason Point-Harding Park                          1.2506   
Norwood                                                                  1.2479   
West Brighton                                                            1.2430   
Longwood                                                            

# **Step 4.3 Run Chi-Square test**
  **Null Hypothesis: There is no association between neighborhoods that have above the average of black and hispanic percentage and have above than average svi scores**

In [204]:
contingency_table = pd.crosstab(evi_svi_df.above_svi_average, evi_svi_df.above_eviction_average)
contingency_table

above_eviction_average  False  True 
above_svi_average                   
False                      29      9
True                       86     63

In [205]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(3.672242587459067),
 np.float64(0.05532569618900214),
 1,
 array([[23.36898396, 14.63101604],
        [91.63101604, 57.36898396]]))

### **P-value is 0.05533 and chi-square is 3.67. We can not refute the null hypothesis. However the p-value is at borderline and the chi-square suggests a small-to-moderate deviation from independence, and therefore it is worthwile to look into it.**

# **conclusion: there is no statistically significant association between the two "above average eviction" and "above average svi scores" neighborhoods.**

In [153]:
a = evi_svi_df[(evi_svi_df['above_svi_average'] == True) & (evi_svi_df['above_eviction_average'] == True)].shape[0]
b = evi_svi_df[(evi_svi_df['above_svi_average'] == True) & (evi_svi_df['above_eviction_average'] == False)].shape[0]
c = evi_svi_df[(evi_svi_df['above_svi_average'] == False) & (evi_svi_df['above_eviction_average'] == True)].shape[0]
d = evi_svi_df[(evi_svi_df['above_svi_average'] == False) & (evi_svi_df['above_eviction_average'] == False)].shape[0]

In [156]:
observed = np.array([[a, b], [c, d]])
observed

array([[63.5, 86.5],
       [ 9.5, 29.5]])

In [157]:
# a += 0.5
# b += 0.5
# c += 0.5
# d += 0.5
# # avoid 0 divisions

In [158]:
odds_ratio = (a * d) / (b * c)
odds_ratio
# extremely strong association between high svi and high evitction rates.

2.2795862488591423

**This means, technically, neighborhoods with above average svi had 2.28 times higher odds of having above average eviction rates compared to low-SVI neighborhoods.** This high odds may be due to all high svi neighborhoods had high evictions, and all high eviction neighborhoods had high svi from the sample.

In [159]:
chi2, p_value, dof, expected = chi2_contingency(observed)
chi2, p_value

(np.float64(3.494209694415174), np.float64(0.061583795661582674))

# **Step 5: We also need a dataframe that has neighborhoods as columns and black and hispanics percentages as rows and contents**

In [173]:
racial_avg_all['black_hispanic_pct'] = racial_avg_all['ep_afam'] + racial_avg_all['ep_hisp']
racial_avg_all

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

In [180]:
evi_bh_df = racial_avg_all[['nta', 'average_year_eviction_count', 'black_hispanic_pct']]
evi_bh_df

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

In [181]:
average_bh_pct = evi_bh_df['black_hispanic_pct'].mean()
average_bh_pct

np.float64(49.541793265928135)

In [182]:
avg_per_nta, average_bh_pct
# evictions and black + hispanic pct

(np.float64(0.9049163589883761), np.float64(49.541793265928135))

In [183]:
evi_bh_df['above_evi_average'] = evi_bh_df['average_year_eviction_count'] > avg_per_nta
evi_bh_df

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

In [187]:
evi_bh_df['above_bh_average'] = evi_bh_df['black_hispanic_pct'] > average_bh_pct
evi_bh_df

nta  \
0                              park-cemetery-etc-Bronx   
1                                 Seagate-Coney Island   
2                                               Corona   
3                    University Heights-Morris Heights   
4                                   Park Slope-Gowanus   
5                        Grymes Hill-Clifton-Fox Hills   
6                                   Morrisania-Melrose   
7                                            Bronxdale   
8                                          Brownsville   
9                                   Claremont-Bathgate   
10                               Westchester-Unionport   
11                   Battery Park City-Lower Manhattan   
12                                       Fordham South   
13                             Turtle Bay-East Midtown   
14                               Oakwood-Oakwood Beach   
15                                      Pelham Parkway   
16                    East Concourse-Concourse Village   
17                                          Mount Hope   
18                    Ft. Totten-Bay Terrace-Clearview   
19                      Melrose South-Mott Haven North   
20                                        East Tremont   
21                                            Flushing   
22                                  Woodlawn-Wakefield   
23                                          Highbridge   
24                                         Kew Gardens   
25                                 Crown Heights South   
26                                      West Concourse   
27                   Central Harlem North-Polo Grounds   
28                           Springfield Gardens North   
29                                          Co-op City   
30                                             Belmont   
31                                   Kew Gardens Hills   
32                               Midtown-Midtown South   
33                                   Crotona Park East   
34                                        Clinton Hill   
35                          Bedford Park-Fordham North   
36                   Prospect Lefferts Gardens-Wingate   
37                                         Hunts Point   
38                 North Riverdale-Fieldston-Riverdale   
39                               Van Cortlandt Village   
40                              West Farms-Bronx River   
41                                  Soundview-Bruckner   
42                            Williamsbridge-Olinville   
43     Soundview-Castle Hill-Clason Point-Harding Park   
44                                             Norwood   
45                                       West Brighton   
46                                            Longwood   
47                                       Dyker Heights   
48                             Briarwood-Jamaica Hills   
49                                       Starrett City   
50                                          Ocean Hill   
51                          Spuyten Duyvil-Kingsbridge   
52                Douglas Manor-Douglaston-Little Neck   
53                                         Murray Hill   
54                                 Kingsbridge Heights   
55                                         Fort Greene   
56                                         Parkchester   
57   Mariner's Harbor-Arlington-Port Ivory-Granitev...   
58                                   East Harlem North   
59                                   East Williamsburg   
60                              East Flatbush-Farragut   
61                                             Jamaica   
62          Hudson Yards-Chelsea-Flatiron-Union Square   
63                                      Bushwick South   
64    DUMBO-Vinegar Hill-Downtown Brooklyn-Boerum Hill   
65                                           Rego Park   
66                                       East New York   
67                                   East Harlem South   
68                                  Marble Hill-I

### **This is the neighborhood evictions + black/hispanic chi-test df:**

# **Run Chi-Square test**
  **Null Hypothesis: There is no association between neighborhoods that have above the average of black and hispanic percentage and have above than average svi scores**

In [188]:
contingency_table = pd.crosstab(evi_bh_df.above_bh_average, evi_bh_df.above_evi_average)
contingency_table

above_evi_average  False  True 
above_bh_average               
False                 80     27
True                  35     45

In [189]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(17.311162196233465),
 np.float64(3.1731742700266107e-05),
 1,
 array([[65.80213904, 41.19786096],
        [49.19786096, 30.80213904]]))

### **P-value is 0.00003.17 (extremely small) and chi-squiare is 17.311. We can refute the null hypothesis. We can say we are confident that high "black+hispanic percetage" boroughs are more likely to have high evictions (or vice versa) or there is a strong association between high BH pct and high eviction rates in the full dataset**

# **conclusion: there is a statistically significant association between the two "above average eviction" and "above average black+hispanic pct" neighborhoods.**

In [190]:
a = evi_bh_df[(evi_bh_df['above_bh_average'] == True) & (evi_bh_df['above_evi_average'] == True)].shape[0]
b = evi_bh_df[(evi_bh_df['above_bh_average'] == True) & (evi_bh_df['above_evi_average'] == False)].shape[0]
c = evi_bh_df[(evi_bh_df['above_bh_average'] == False) & (evi_bh_df['above_evi_average'] == True)].shape[0]
d = evi_bh_df[(evi_bh_df['above_bh_average'] == False) & (evi_bh_df['above_evi_average'] == False)].shape[0]

In [191]:
observed = np.array([[a, b], [c, d]])
observed

array([[45, 35],
       [27, 80]])

In [193]:
odds_ratio = (a * d) / (b * c)
odds_ratio

3.8095238095238093

**This means, technically, neighborhoods with above average b+h pct had 3.81 times higher odds of having above average eviction rates compared to low-bh-pct neighborhoods in nyc.** This high odds may be due to all high bh pct neighborhoods had high evictions, and all high eviction neighborhoods had high bh pct from the sample.

In [194]:
chi2, p_value, dof, expected = chi2_contingency(observed)
chi2, p_value
# 11.32 is a strong deviation from expected undert the null hypothesis

(np.float64(17.311162196233465), np.float64(3.1731742700266107e-05))

### **Quick extreme cases**

In [207]:
evi_svi_df_top_10 = evi_svi_df.sort_values('average_year_eviction_count', ascending=False).head(10)
evi_svi_df_bottom_10 = evi_svi_df.sort_values('average_year_eviction_count', ascending=False).tail(10)
evi_svi_df_extremes = pd.concat([evi_svi_df_top_10, evi_svi_df_bottom_10])
evi_svi_df_extremes

average_year_eviction_count  \
nta                                                                       
park-cemetery-etc-Bronx                                          6.4667   
Seagate-Coney Island                                             4.3904   
Corona                                                           4.2597   
University Heights-Morris Heights                                3.5051   
Park Slope-Gowanus                                               2.3725   
Grymes Hill-Clifton-Fox Hills                                    2.3360   
Morrisania-Melrose                                               2.2550   
Bronxdale                                                        2.2396   
Brownsville                                                      2.0909   
Claremont-Bathgate                                               1.9783   
Greenpoint                                                       0.2531   
Charleston-Richmond Valley-Tottenville                           0.2516   
Great Kills                                                      0.2488   
Annadale-Huguenot-Prince's Bay-Eltingville                       0.2471   
Westerleigh                                                      0.2471   
Maspeth                                                          0.2467   
Arden Heights                                                    0.2235   
Glen Oaks-Floral Park-New Hyde Park                              0.2235   
Rossville-Woodrow                                                0.2235   
park-cemetery-etc-Brooklyn                                       0.2000   

                                            rpl_themes  above_svi_average  \
nta                                                                         
park-cemetery-etc-Bronx                         0.9925               True   
Seagate-Coney Island                            0.9911               True   
Corona                                          0.9632               True   
University Heights-Morris Heights               0.9979               True   
Park Slope-Gowanus                              0.6327              False   
Grymes Hill-Clifton-Fox Hills                   0.9444               True   
Morrisania-Melrose                              0.9965               True   
Bronxdale                                       0.9909               True   
Brownsville                                     0.9933               True   
Claremont-Bathgate                              0.9971               True   
Greenpoint                                      0.6370              False   
Charleston-Richmond Valley-Tottenville          0.4412              False   
Great Kills                                     0.5291              False   
Annadale-Huguenot-Prince's Bay-Eltingville      0.5460              False   
Westerleigh                                     0.8489               True   
Maspeth                                         0.8709               True   
Arden Heights                                   0.5333              False   
Glen Oaks-Floral Park-New Hyde Park             0.7721              False   
Rossville-Woodrow                               0.5606              False   
park-cemetery-etc-Brooklyn                      0.9427               True   

                                            above_eviction_average  
nta                                                                 
park-cemetery-etc-Bronx                                       True  
Seagate-Coney Island                                          True  
Corona                                                        True  
University Heights-Morris Heights                             True  
Park Slope-Gowanus                                            True  
Grymes Hill-Clifton-Fox Hills                                 True  
Morrisania-Melrose                                            True  
Bronxdale                                                     True  
Brownsvi

In [211]:
contingency_table = pd.crosstab(evi_svi_df_extremes.above_svi_average, evi_svi_df_extremes.above_eviction_average)
contingency_table

above_eviction_average  False  True 
above_svi_average                   
False                       7      1
True                        3      9

In [212]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(5.208333333333334),
 np.float64(0.022478873366125265),
 1,
 array([[4., 4.],
        [6., 6.]]))

### **P-value is 0.022 (extremely small) and chi-squiare is 5.208. We can refute the null hypothesis. We can say we are confident that high "svi" boroughs are more likely to have high evictions (or vice versa) or there is a strong association between high SVI and high eviction rates in the full dataset**

# **conclusion: there is a statistically significant association between the two "above average eviction" and "above average black+hispanic pct" neighborhoods.**

In [214]:
a = evi_svi_df_extremes[(evi_svi_df_extremes['above_svi_average'] == True) & (evi_svi_df_extremes['above_eviction_average'] == True)].shape[0]
b = evi_svi_df_extremes[(evi_svi_df_extremes['above_svi_average'] == True) & (evi_svi_df_extremes['above_eviction_average'] == False)].shape[0]
c = evi_svi_df_extremes[(evi_svi_df_extremes['above_svi_average'] == False) & (evi_svi_df_extremes['above_eviction_average'] == True)].shape[0]
d = evi_svi_df_extremes[(evi_svi_df_extremes['above_svi_average'] == False) & (evi_svi_df_extremes['above_eviction_average'] == False)].shape[0]

In [215]:
observed = np.array([[a, b], [c, d]])
observed

array([[9, 3],
       [1, 7]])

In [216]:
odds_ratio = (a * d) / (b * c)
odds_ratio

21.0

**This means, technically, neighborhoods with above average b+h pct had 21 times higher odds of having above average eviction rates compared to low-bh-pct neighborhoods in nyc.** This high odds may be due to all high bh pct neighborhoods had high evictions, and all high eviction neighborhoods had high bh pct from the sample.

In [217]:
evi_bh_df_top_10 = evi_bh_df.sort_values('average_year_eviction_count', ascending=False).head(10)
evi_bh_df_bottom_10 = evi_bh_df.sort_values('average_year_eviction_count', ascending=False).tail(10)
evi_bh_df_extremes = pd.concat([evi_bh_df_top_10, evi_bh_df_bottom_10])
evi_bh_df_extremes

nta  average_year_eviction_count  \
0                       park-cemetery-etc-Bronx                       6.4667   
1                          Seagate-Coney Island                       4.3904   
2                                        Corona                       4.2597   
3             University Heights-Morris Heights                       3.5051   
4                            Park Slope-Gowanus                       2.3725   
5                 Grymes Hill-Clifton-Fox Hills                       2.3360   
6                            Morrisania-Melrose                       2.2550   
7                                     Bronxdale                       2.2396   
8                                   Brownsville                       2.0909   
9                            Claremont-Bathgate                       1.9783   
177                                  Greenpoint                       0.2531   
178      Charleston-Richmond Valley-Tottenville                       0.2516   
179                                 Great Kills                       0.2488   
180  Annadale-Huguenot-Prince's Bay-Eltingville                       0.2471   
181                                 Westerleigh                       0.2471   
182                                     Maspeth                       0.2467   
183                               Arden Heights                       0.2235   
184         Glen Oaks-Floral Park-New Hyde Park                       0.2235   
185                           Rossville-Woodrow                       0.2235   
186                  park-cemetery-etc-Brooklyn                       0.2000   

     black_hispanic_pct  above_evi_average  above_bh_average  
0               81.5000               True              True  
1               39.5544               True             False  
2               83.1637               True              True  
3               93.7760               True              True  
4               23.8314               True             False  
5               47.5007               True             False  
6               94.6076               True              True  
7               81.5607               True              True  
8               89.4267               True              True  
9               93.7809               True              True  
177             18.2000              False             False  
178             10.7581              False             False  
179             13.8659              False             False  
180             12.1588              False             False  
181             35.7941              False             False  
182             37.7150              False             False  
183             13.6000              False             False  
184             17.9941              False             False  
185             10.5118              False             False  
186             35.2000              False             False

In [220]:
contingency_table = pd.crosstab(evi_bh_df_extremes.above_bh_average, evi_bh_df_extremes.above_evi_average)
contingency_table

above_evi_average  False  True 
above_bh_average               
False                 10      3
True                   0      7

In [221]:
chi2, p_value, dof, expected = chi2_contingency(contingency_table)
chi2, p_value, dof, expected

(np.float64(7.912087912087912),
 np.float64(0.004910556162172649),
 1,
 array([[6.5, 6.5],
        [3.5, 3.5]]))

### **P-value is 0.0049 (extremely small) and chi-squiare is 7.812. We can refute the null hypothesis. We can say we are confident that high "black+hispanic" boroughs are more likely to have high evictions (or vice versa) or there is a strong association between high black+hispanic and high eviction rates in the full dataset**

# **conclusion: there is a statistically significant association between the two "above average eviction" and "above average black+hispanic pct" neighborhoods.**

In [224]:
a = evi_bh_df_extremes[(evi_bh_df_extremes['above_bh_average'] == True) & (evi_bh_df_extremes['above_evi_average'] == True)].shape[0]
b = evi_bh_df_extremes[(evi_bh_df_extremes['above_bh_average'] == True) & (evi_bh_df_extremes['above_evi_average'] == False)].shape[0]
c = evi_bh_df_extremes[(evi_bh_df_extremes['above_bh_average'] == False) & (evi_bh_df_extremes['above_evi_average'] == True)].shape[0]
d = evi_bh_df_extremes[(evi_bh_df_extremes['above_bh_average'] == False) & (evi_bh_df_extremes['above_evi_average'] == False)].shape[0]

In [225]:
observed = np.array([[a, b], [c, d]])
observed

array([[ 7,  0],
       [ 3, 10]])

In [226]:
a += 0.5
b += 0.5
c += 0.5
d += 0.5

In [227]:
odds_ratio = (a * d) / (b * c)
odds_ratio

45.0

**This means, technically, neighborhoods with above average b+h pct had 45 times higher odds of having above average eviction rates compared to low-bh-pct neighborhoods in nyc.** This high odds may be due to all high bh pct neighborhoods had high evictions, and all high eviction neighborhoods had high bh pct from the sample.